messagecatalog/spec.md

```diff
@@ -0,0 +1,1475 @@
+
+Structurally and functionally, this message catalog API is very similar to the
+CNCF Schema Registry API, and it is reasonable and desirable for those APIs to
+be collocated in the same service and on the same endpoint.
+
+The Message Catalog and Schema Registry data formats and APIs exist in parallel
+because they serve different purposes. The Schema Registry is a general purpose
+store for serialization and validation schemas documents for structured data
+that is useful for describing message and event payloads, but also in all other
+scenarios where structured data is handled. For instance, a data structure might
+be sent to a system inside of a message but might subsequently be stored inside
+a data lake without the message envelope. The Message Catalog is more
+constrained and focused on message definitions, which covers both payload and
+metadata of the message. In environments where data structures are shared across
+multiple services and applications, the Schema Registry and the Message Catalog
+might be managed separately, with Message Catalog entries always referring to
+data type definitions held in the Schema Registry. In simple applications, all
+data type definitions for payloads MAY be held inline inside the Message
+Catalog.
+
+The message definition entries in the catalog contain some fixed metadata
+information. That fixed information MUST uniquely identify the type of message
+and depends on the particular message information model. For example, for CNCF
+CloudEvents message definitions, the "type" attribute is REQUIRED as fixed
+information for each entry.
+
+- Message definition group: A named collection of message definitions. Each
+  group holds a logically related set of message definitions, typically managed
+  by a single entity, belonging to a particular application and/or having a
+  shared access control management scope.
+
+- Message definition: A document describing the metadata and structure of
+  messages and events.
+
+- Message definition version: A specific version of such a definition. All
+  documents that are stored and retrieved through the API are such versions. The
+  identifying criteria of a definition MUST be identical across versions.
+
+A message definition group, including all definitions and its versions, can be
+stored and shared in a single document. A folder in a git repository containing
+such documents is also a catalog in the sense of this specification.
+
+For interoperability and sharing design-time definitions across system
+boundaries, conformance to a common data model and document format is most
+important and it is assumes that the authoritative source for many definitions
+will into be a document under source control that is then shared at runtime via
+the common API defined here.
+
+## 2. Message Catalog Elements
+
+This section describes the elements enumerated in the introduction.
+
+All data types used in this section are defined in the CNCF CloudEvents
```

Comment:
we should add a link to the CE spec so people know exactly which section to look at.